src/control_receiver.py

```diff
@@ -94,38 +94,42 @@ def keyboard_handler(self, control, value):
                 self.motors.keyboard_speed = max(127, speed - 128)
                 self.pipe.send(["SYNC_SPEED", "kb", self.motors.keyboard_speed])
 
-    def save_config(self, cfg):
+    def save_config(self, cfg, filename):
         # Rolling backups
         backup_dir = "src/configs/sights/backup/"
-        name = os.path.basename(self.config_file)
-        # Find existing backups for this config file
-        for file in sorted(os.listdir(backup_dir), reverse=True):
-            if file.startswith(f"{name}.backup."):
-                # Get backup ID
-                id = int(file[-1])
-                # Remove oldest backup
-                if id == 5:
-                    os.remove(backup_dir + file)
-                else:
-                    # Add 1 to the rest of the backup IDs
-                    new_id = str(id+1)
-                    os.rename(backup_dir + file, backup_dir + name + ".backup." + new_id)
-        # Save the previous config as a new backup
-        os.rename(self.config_file, backup_dir + name + ".backup.0")
+        # Use current config name if no new name is provided
+        name = os.path.basename(self.config_file) if filename == "" else filename
+        config_path = os.path.dirname(self.config_file) + "/" + name
 
-        # Save new config to file
-        with open(self.config_file, 'w') as f:
-            f.write(cfg)
-        # Reload config 
-        self.config = json.load(open(self.config_file))
-        self.logger.info("Saved configuration file to " + self.config_file)
+        if os.path.exists(config_path):
+            # File exists
+            # Find existing backups for this config file
+            for file in sorted(os.listdir(backup_dir), reverse=True):
+                if file.startswith(f"{name}.backup."):
+                    # Get backup ID
+                    id = int(file[-1])
+                    # Remove oldest backup
+                    if id == 5:
+                        os.remove(backup_dir + file)
+                    else:
+                        # Add 1 to the rest of the backup IDs
+                        new_id = str(id + 1)
+                        os.rename(backup_dir + file, backup_dir + name + ".backup." + new_id)
+            # Save the previous config as a new backup
+            os.rename(config_path, backup_dir + name + ".backup.0")
+
+            # Save new config to file
+            with open(config_path, 'w') as f:
+                f.write(cfg)
+            # Reload config
+            self.config = json.load(open(config_path))
```

Comment:
Does this reload do anything useful? As far as I know, a restart of the service is still required. Should it be removed?

Reply:
Good point. I originally envisioned reloading stuff manually within the script, but doing it by restarting the service is so much simpler and easier. 

Let's remove that to ensure that functionality is consistent.

Reply:
That's a great idea! I thought you were going for something like that. It does sound like a lot of work though - just removed it in 025040b3cf9af24b6b586567ae53b8d3e50fc87b.